var/spack/repos/builtin/packages/yafyaml/package.py

```diff
@@ -60,6 +62,35 @@ class Yafyaml(CMakePackage):
         msg="yaFyaml only works with the Fujitsu compiler from 1.3.0 onwards",
     )
 
+    # GCC 13.3 and higher only work with yafyaml 1.4.0 onwards
+    # First we can check if the spec is gcc@13.3...
+    conflicts("%gcc@13.3:", when="@:1.3.0", msg="GCC 13.3+ only works with yafyaml 1.4.0 onwards")
+
+    # ...but if it is not (say apple-clang with gfortran as a fc), there is
```

Comment:
I am only seeing this now. There is a much simpler way to achieve the same: https://github.com/spack/spack/blob/7efbad0d812f17c2dcffe32649d9c14db82a69dd/var/spack/repos/builtin/packages/esmf/package.py#L219

Reply:
Ahh. I didn't know about `compiler_environment()`. I'll take a look at updating the code...